hollowverseComCloudFront/hollowverseComCloudFront.md

```diff
@@ -0,0 +1,37 @@
+# hollowverse.com CloudFront distribution
+
+```mermaid
+graph TD
+  start(Receive request) --> aetvr["Assign beta, master, or other<br>environment to request<br><br>(assignEnvironmentToViewerRequest)"]
+  aetvr --> isCached{Is content<br>cached for<br>assigned<br>environment?}
+  isCached --> |Yes| cachedContent[Get cached content]
+  cachedContent --> response(Send response)
+  isCached --> |No| routeRequestToOrigin["Determine the server<br>for the assigned<br>environment<br><br>(routeRequestToOrigin)"]
+  routeRequestToOrigin --> whichEnv{Which<br>environment<br>was<br>selected?}
+  whichEnv -- beta --> beta[Retrieve content from<br>beta environment]
+  whichEnv -- master --> master[Retrieve content from<br>master environment]
+  whichEnv -- other --> other[Retrieve content from<br>whatever environment]
+  beta --> setHeaders["Set headers on response<br>before sending it back<br><br>(setHeadersOnOriginResponse)"]
+  master --> setHeaders
+  other --> setHeaders
+  setHeaders --> cache[Add response to cache<br>for subsequent requests]
+  cache --> response
+```
+
+When CloudFront receives the request for the page, it executes the Lambda@Edge assigned to the `viewer-request` stage. This is [`assignEnvironmentToViewerRequest`](https://github.com/hollowverse/route-request/blob/master/src/assignEnvironmentToViewerRequest.ts).
+
+`assignEnvironmentToViewerRequest` modifies the incoming user request and returns the modifies request [TK modifies the request based on what? Why does it modify the request?].
```

Comment:
@forabi what does `assignEvironmentToViewerRequest` modify the request based on, and why does it modify the request?

Reply:
@wholesomedev As the name implies, it modifies the request to add an environment, specifically to add the `x-hollowverse-assigned-environment` header (whose value is a random environment picked based on weights if no `env` cookie exists in the request, or the value of `env` cookie if it exists and it is a valid environment).